hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -3197,21 +3198,26 @@ protected String getPrunedDiscriminatorPredicate(
 		}
 
 		if ( containsNotNull ) {
-			final StringBuilder sb = new StringBuilder();
-			String lhs;
+			final StringBuilder sb = new StringBuilder( " or " );
```

Comment:
Just a rough estimation to avoid some resizes:

```suggestion
			final StringBuilder sb = new StringBuilder( 70 + discriminatorSQLValues.size() * 10 ).append( " or " );
```